src/CoinParamUtils.cpp

```diff
@@ -602,9 +726,6 @@ int matchParam(const CoinParamVec &paramVec, std::string name,
   shortCnt = 0;
 
   for (int i = 0; i < vecLen; i++) {
-    CoinParam *param = paramVec[i];
```

Comment:
if params are now != NULL, we might be able to use std::vector<T> instead of std::vector<T*>, which would
1.) avoid accidential NULL
2.) new'ing every element in the vector (as seen in CbcParameters)
3.) deleting every element in the vector (as seen in CbcParameters).

vector[] returns references, so we could kill a lot of pointers and questions "can it be NULL" and "who owns it" are avoided/enforced by the language. I'm currently evaluating if it is actual doable or if there is a piece of code that needs a pointer

Reply:
Well, yes, I went around and about on all of this stuff when I was thinking through the whole thing and if you look at the history of the PR in CLP, you will see that I did do exactly as you said initially. At some point, I ran into an issue with doing that and switched everything back again. If I were a professional and this was my day job :), I would have more carefully documented why I made the switch. But you can see from the comment in [this commit](https://github.com/coin-or/Clp/commit/867718cc202c787466e3c5a41e3ef95c3f6ab615#diff-25a6634263c1b1f6fc4697a04e2b9904ea4b042a89af59dc93ec1f5d44848a26) that I somehow decided we need `std::vector<T*>`.

With all that said, once things are up and running, I'm happy to discuss any of these sort of design decisions and revisit them to fine-tune. I was working on this in a vacuum initially. It's nice to have some sort of sounding board.